server/routes/restaurant.js

```diff
@@ -105,15 +105,26 @@ router.get('/searchRestaurants/:latitude/:longitude/:radius', async (req, res) =
                     const menu_find_result = await databaseMaster.dbOp('find', 'MenuDetails', { query: { restaurantId: fsq_id } });
                     const placeDetails = response1.data;
 
-                    //iterate through the categories array to retrieve the cuisines
+                    //iterate through the categories array to retrieve the cuisines and restaurantTypes
                     const categories_array = placeDetails.categories;
-                    const cuisine = [];
-                    cuisine_options = ["Chinese", "Australian", "Thai", "Asian", "Italian", "Greek", "Mexican", "Japanese", "Vietnamese", "Korean", "American", "French", "Turkish", "Indian"]
+                    const cuisineType_array = [];
+                    const restaurantType_array = [];
+                    const cuisineTypes = ["Chinese", "Australian", "Thai", "Asian", "Italian", "Greek", "Mexican", "Japanese", "Vietnamese", "Korean", "American", "French", "Turkish", "Indian", "Malay", "Korean BBQ", "Mediterranean"];
+                    const restaurantTypes = ["Bakery", "Bar", "Pub", "Café", "Coffee Shop", "Halal", "Vegan and Vegetarian", "Desserts", "Donuts", "Steakhouse", "Diner", "Fish and Chips Shop", "Ice Cream", "Tea Room", "Gelato", "Pizza", "Hotel Bar", "Wine Bar", "Beer Bar"];
                     for (i = 0; i < categories_array.length; i++)
                     {
-                        if(cuisine_options.includes(categories_array[i].short_name))
+                        if(cuisineTypes.includes(categories_array[i].short_name))
                         {
-                            cuisine.push(categories_array[i].short_name);
+                            //retrieve the cuisine name and associated icon png at size 64 pixels
+                            const cuisine_icon_url = categories_array[i].icon.prefix + "bg_64" + categories_array[i].icon.suffix
+                            cuisineType_array.push({"cuisineType": categories_array[i].short_name, "icon" : cuisine_icon_url});
+                        }  
+
+                        else if(restaurantTypes.includes(categories_array[i].short_name))
```

Comment:
Just my preference, but would prefer to add every category besides `cuisineTypes` into the `restaurantTypes` (wouldn't restrict it with `restaurantTypes[]` as we're missing some good ones - there's a lot of categories haha)

Reply:
Yeh the categories are quite broad and numerous, only problem with your solution is that the RestaurantType might include categories not related to food and dining.  This is as some places can act both as restaurants and entertainment i.e. a karaoke bar, hotel with a restaurant, music venue that has a bar etc

Reply:
Yeah that should be okay, filtering can also be done in the frontend - if a restaurant has extra entertainment like karaoke, I think it would be cool to have it in the restaurant details too

Reply:
Resolved in commit https://github.com/abrarbarbhuyia/2024SIS_Team5-/pull/26/commits/ad7221fb77fda22e95521b92051dd084c7cd7aac